File: packages/contracts/src/interfaces/enums.ts
/**
 * 花卉品类枚举，用于植株档案分类与筛选。
 */
export enum FlowerCategory {
  Succulent = 'succulent',
  Herbaceous = 'herbaceous',
  Woody = 'woody',
  Hydroponic = 'hydroponic',
  Vine = 'vine',
}

/**
 * 植株摆放位置枚举，用于天气和养护建议判断。
 */
export enum FlowerPlacement {
  IndoorBalcony = 'indoor_balcony',
  OutdoorOpenAir = 'outdoor_open_air',
  IndoorShade = 'indoor_shade',
}

/**
 * 养护难度枚举，用于新手提示和分类筛选。
 */
export enum FlowerCareDifficulty {
  Easy = 'easy',
  Medium = 'medium',
  Hard = 'hard',
}

/**
 * 养护记录操作类型枚举。
 */
export enum RecordActionType {
  Watering = 'watering',
  Fertilizing = 'fertilizing',
  Pruning = 'pruning',
  Repotting = 'repotting',
  PestControl = 'pest_control',
  LeafCleaning = 'leaf_cleaning',
  LightAdjustment = 'light_adjustment',
}

/**
 * 天气状态枚举，用于前后端统一映射天气文案。
 */
export enum WeatherCondition {
  Sunny = 'sunny',
  Cloudy = 'cloudy',
  Overcast = 'overcast',
  Rainy = 'rainy',
  Storm = 'storm',
  Snow = 'snow',
  Foggy = 'foggy',
  Hazy = 'hazy',
}

/**
 * 会员权益项枚举。
 */
export enum MemberBenefitType {
  UnlimitedAiAdvice = 'unlimited_ai_advice',
  NoWatermark = 'no_watermark',
  CloudBackup = 'cloud_backup',
  AllThemes = 'all_themes',
  AdvancedTheme = 'advanced_theme',
  GrowthPoster = 'growth_poster',
  TripCarePlan = 'trip_care_plan',
  AdFree = 'ad_free',
}

/**
 * 会员套餐枚举。
 */
export enum MemberPackageType {
  Monthly = 'monthly',
  Yearly = 'yearly',
  Lifetime = 'lifetime',
}

/**
 * 会员状态枚举。
 */
export enum MemberStatus {
  Inactive = 'inactive',
  Active = 'active',
  Expired = 'expired',
  GracePeriod = 'grace_period',
}

/**
 * 会员支付渠道枚举。
 */
export enum MemberPaymentChannel {
  MpWeixin = 'mp_weixin',
  H5QrCode = 'h5_qr_code',
}

/**
 * 会员支付订单状态枚举。
 */
export enum MemberPaymentStatus {
  Pending = 'pending',
  Paid = 'paid',
  Expired = 'expired',
  Closed = 'closed',
}

/**
 * 用户反馈处理状态枚举。
 */
export enum FeedbackStatus {
  /** 待 AI 审核 */
  Pending = 'pending',
  /** AI 审核通过，已公开 */
  Approved = 'approved',
  /** AI 审核驳回 */
  Rejected = 'rejected',
  /** 管理员已审阅 */
  Reviewed = 'reviewed',
  /** 已归档 */
  Archived = 'archived',
}

/**
 * 登录来源枚举。
 */
export enum UserLoginType {
  AnonymousLocal = 'anonymous_local',
  WechatMiniProgram = 'wechat_mini_program',
  H5PhoneCode = 'h5_phone_code',
}

/**
 * 用户状态枚举。
 */
export enum UserStatus {
  Normal = 'normal',
  Disabled = 'disabled',
}

File: services/ai-agent/app/constants/enums.py
"""Mirror of the existing contracts enums from @florist/contracts."""

from enum import StrEnum


class FlowerCategory(StrEnum):
    SUCCULENT = "succulent"
    HERBACEOUS = "herbaceous"
    WOODY = "woody"
    HYDROPONIC = "hydroponic"
    VINE = "vine"


class FlowerPlacement(StrEnum):
    INDOOR_BALCONY = "indoor_balcony"
    OUTDOOR_OPEN_AIR = "outdoor_open_air"
    INDOOR_SHADE = "indoor_shade"


class FlowerCareDifficulty(StrEnum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"


class FlowerHealthStatus(StrEnum):
    WATERING_NEEDED = "watering-needed"
    HEALTHY = "healthy"
    DORMANT = "dormant"
    FERTILIZING_NEEDED = "fertilizing-needed"


class RecordActionType(StrEnum):
    WATERING = "watering"
    FERTILIZING = "fertilizing"
    PRUNING = "pruning"
    REPOTTING = "repotting"
    PEST_CONTROL = "pest_control"
    LEAF_CLEANING = "leaf_cleaning"
    LIGHT_ADJUSTMENT = "light_adjustment"


class WeatherCondition(StrEnum):
    SUNNY = "sunny"
    CLOUDY = "cloudy"
    OVERCAST = "overcast"
    RAINY = "rainy"
    STORM = "storm"
    SNOW = "snow"
    FOGGY = "foggy"
    HAZY = "hazy"


ACTION_LABELS: dict[RecordActionType, str] = {
    RecordActionType.WATERING: "浇水",
    RecordActionType.FERTILIZING: "施肥",
    RecordActionType.PRUNING: "修剪",
    RecordActionType.REPOTTING: "换盆",
    RecordActionType.PEST_CONTROL: "除虫",
    RecordActionType.LEAF_CLEANING: "擦叶",
    RecordActionType.LIGHT_ADJUSTMENT: "调整光照",
}

CATEGORY_LABELS: dict[FlowerCategory, str] = {
    FlowerCategory.SUCCULENT: "多肉",
    FlowerCategory.HERBACEOUS: "草本",
    FlowerCategory.WOODY: "木本",
    FlowerCategory.HYDROPONIC: "水培",
    FlowerCategory.VINE: "藤蔓",
}

PLACEMENT_LABELS: dict[FlowerPlacement, str] = {
    FlowerPlacement.INDOOR_BALCONY: "室内阳台",
    FlowerPlacement.OUTDOOR_OPEN_AIR: "户外露天",
    FlowerPlacement.INDOOR_SHADE: "室内阴面",
}
